### PR TITLE
[VBLOCKS-2081] Persist outgoing client identity

### PR DESCRIPTION
**Pull request by mhuynh5757**
## Submission Checklist

 - [x] The `CHANGELOG.md` reflects any **feature**, **bug fixes**, or **known issues** made in the source code
 - [x] New unit tests and integration tests have beed added
 - [x] Code coverage is more or equal to the previous coverage
 - [x] A visual inspection of the `Files changed` tab was made prior to submitting the pull request ensuring the style guide was followed
 - [x] [CI pipeline](https://app.circleci.com/pipelines/github/twilio/twilio-voice-react-native-app) is green
 - [ ] Included screenshot if necessary

## Description

This PR persists the outgoing call parameters for UX purposes.

### Breakdown

- Use AsyncStorage library to persist information about outgoing calls between JS runtimes.
- Slight refactor of the custom parameters info submember of the call entity type.

## Validation

- Unit tests.
- Manual testing on Android.

## Additional Notes

N/A

## Contributing to Twilio

> All third-party contributors acknowledge that any contributions they provide will be made under the same open-source license that the open-source project is provided under.

- [x] I acknowledge that all my contributions will be made under the project's license.

### CODE REVIEW

File: app/src/store/bootstrap.ts [hunk 1]
```diff
@@ -1,3 +1,4 @@
+import AsyncStorage from '@react-native-async-storage/async-storage';
```
Comment (kpchoy) [inline app/src/store/bootstrap.ts:1]: Why use this instead of just using the redux store?

File: app/src/store/bootstrap.ts [hunk 2]
```diff
@@ -209,9 +210,25 @@ export const bootstrapCalls = createTypedAsyncThunk<
...
+    // Get all the call sids stored in async storage.
+    const storedCallSids = new Set(await AsyncStorage.getAllKeys());
     for (const call of calls.values()) {
       await dispatch(handleCall({ call }));
+
+      // If the call is still active, the native layer will still have it
+      // cached.
+      const callSid = call.getSid();
+      if (callSid) {
+        // Mark a call as still active, and therefore keep it in async storage.
```
Comment (kpchoy) [inline app/src/store/bootstrap.ts:222]: keep by using `delete` ?

File: app/src/store/bootstrap.ts (hunk 1 above)
Comment (mhuynh5757) [reply app/src/store/bootstrap.ts:1]: `AsyncStorage` persists through JS runtimes, whereas the Redux store is wiped and reset every time the app is launched.

File: app/src/store/bootstrap.ts (hunk 2 above)
Comment (mhuynh5757) [reply app/src/store/bootstrap.ts:222]: I suppose the comment is a bit strangely worded, but the idea is that if a call remains (i.e. it is _not_ deleted) then it is stale. We we mark it as active (not stale) by deleting it from the set of stale calls.

**Review by kpchoy — APPROVED** [review]

**Pull request merged**